### PR TITLE
Linear motion compensator v2

### PR DESCRIPTION
**Pull request by Ollieollieolllie**

### CODE REVIEW

File: @AH-64D Apache Longbow/Addons/fza_ah64_ihadss/functions/fn_draw.sqf [hunk 1]
```diff
@@ -220,6 +227,12 @@ if (cameraView == "GUNNER" && player == gunner _heli) then {
         ((uiNameSpace getVariable "fza_ah64_laseit") displayCtrl 701) ctrlSetText "\fza_ah64_US\tex\HDU\Apache_LaserOn.paa";
         ((uiNameSpace getVariable "fza_ah64_laseit") displayCtrl 701) ctrlSetTextColor[(_hduColour select 1), (_hduColour select 1), (_hduColour select 1), 1];
     };
+    //LSC SYMBOLOGY FOR GUNNER
+    if (_heli getvariable "fza_ah64_LmcActive") then {
+        _LMCLayer cutrsc["fza_ah64_LMC", "PLAIN", 0.01, false];
+        ((uiNameSpace getVariable "fza_ah64_LMC") displayCtrl 703) ctrlSetText "\fza_ah64_US\tex\HDU\TADSLMC_co.paa";
```
Comment (mattysmith22) [inline @AH-64D Apache Longbow/Addons/fza_ah64_ihadss/functions/fn_draw.sqf:233]: Does this symbology really need its own RSC layer? Can it not be integrated with the rest?

File: @AH-64D Apache Longbow/Addons/fza_ah64_ihadss/functions/fn_draw.sqf [hunk 2]
```diff
@@ -244,6 +257,7 @@ if (cameraView == "GUNNER" && player == gunner _heli) then {
     ctrlSetTextColor[1, 1, 1, 1];
     ((uiNameSpace getVariable "fza_ah64_raddisp") displayCtrl 130) ctrlSetText "\fza_ah64_US\tex\HDU\ihadss.paa"; //TEST
     ((uiNameSpace getVariable "fza_ah64_laseit")  displayCtrl 701) ctrlSetText "";
+    ((uiNameSpace getVariable "fza_ah64_laseit")  displayCtrl 703) ctrlSetText "";
```
Comment (mattysmith22) [inline @AH-64D Apache Longbow/Addons/fza_ah64_ihadss/functions/fn_draw.sqf:260]: It's this the wrong rsc layer?

File: @AH-64D Apache Longbow/Addons/fza_ah64_ihadss/functions/fn_linearMotionCompensator.sqf [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    "a3\ui_f\data\Map\Markers\Military\circle_CA.paa",
+    [1, 1, 1, 1],
+    positionCameraToWorld [_inputX, _inputY, 10],
+    1, 1, 0
+];
+#endif
+
+//AUTO RANGING USING ATL ON 2D Model
+private _TadsPosition  = _heli modelToWorldVisualWorld (_heli selectionPosition "laserEnd");
+private _TadsAimPos    = _heli modelToWorldVisualWorld (_heli selectionPosition "laserBegin");
```
Comment (mattysmith22) [inline @AH-64D Apache Longbow/Addons/fza_ah64_ihadss/functions/fn_linearMotionCompensator.sqf:52]: Wrong case

File: @AH-64D Apache Longbow/Addons/fza_ah64_ihadss/functions/fn_linearMotionCompensator.sqf [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+    _heli - The helicopter to act on
+
+Returns:
+
+Examples:
+
+Author:
+    Snow(Dryden), Ampersand
+---------------------------------------------------------------------------- */
+#define INPUT_SENS 0.05 // Tweak this for responsiveness? - Ampersand
```
Comment (mattysmith22) [inline @AH-64D Apache Longbow/Addons/fza_ah64_ihadss/functions/fn_linearMotionCompensator.sqf:17]: We could also allow for adjustment using a CBA setting slider?

File: @AH-64D Apache Longbow/Addons/fza_ah64_ihadss/functions/fn_linearMotionCompensator.sqf [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+#endif
+
+//AUTO RANGING USING ATL ON 2D Model
+private _TadsPosition  = _heli modelToWorldVisualWorld (_heli selectionPosition "laserEnd");
+private _TadsAimPos    = _heli modelToWorldVisualWorld (_heli selectionPosition "laserBegin");
+private _tadsDirection = (_TadsPosition vectorFromTo _TadsAimPos) vectorMultiply 50000;
+_tadsDirection call CBA_fnc_vect2Polar Params ["","","_elevation"];
+private _autorange = [(_TadsPosition)#2 /sin(-_elevation),0,50000] call BIS_fnc_clamp;
+
+private _range = -1;
```
Comment (mattysmith22) [inline @AH-64D Apache Longbow/Addons/fza_ah64_ihadss/functions/fn_linearMotionCompensator.sqf:57]: The range source logic here looks good, but I would prefer for us to have a separate function that does this for reuse

**Review by mattysmith22 — CHANGES_REQUESTED** [review]
Great start! Some suggestions. 

You also mentioned that you need to make the LMC use delta time

File: @AH-64D Apache Longbow/Addons/fza_ah64_ihadss/functions/fn_linearMotionCompensator.sqf (hunk 5 above)
Comment (Ollieollieolllie) [reply @AH-64D Apache Longbow/Addons/fza_ah64_ihadss/functions/fn_linearMotionCompensator.sqf:57]: Will be left for a seperate PR

File: @AH-64D Apache Longbow/Addons/fza_ah64_ihadss/functions/fn_linearMotionCompensator.sqf (hunk 3 above)
Comment (Ollieollieolllie) [reply @AH-64D Apache Longbow/Addons/fza_ah64_ihadss/functions/fn_linearMotionCompensator.sqf:52]: fixed

File: @AH-64D Apache Longbow/Addons/fza_ah64_ihadss/functions/fn_draw.sqf (hunk 1 above)
Comment (Ollieollieolllie) [reply @AH-64D Apache Longbow/Addons/fza_ah64_ihadss/functions/fn_draw.sqf]: moved

File: @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/core/fn_coreControlHandle.sqf [hunk 6]
```diff
@@ -238,6 +238,11 @@ if (_value) then {
                 _heli setvariable ["fza_ah64_tadsZoom", 0];
             };
         };
+        case "vehLockTurretView": {
+            if (player != Gunner _heli) exitWith {};
```
Comment (mattysmith22) [inline @AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/core/fn_coreControlHandle.sqf:242]: `gunner`

**Review by mattysmith22 — APPROVED** [review]

**Comment by Ollieollieolllie** [comment]
think I'm gonna need brads sign off first, I don't think he's tested it yet

**Review by mattysmith22 — CHANGES_REQUESTED** [review]
Scaling required based on FOV - zoomed in are too sensitive

**Pull request merged**